### PR TITLE
added autocorrect to LeadingWhiteSpace

### PR DESCRIPTION
**Pull request by masters3d**

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift [hunk 1]
```diff
@@ -9,7 +9,23 @@
 import Foundation
 import SourceKittenFramework
 
-public struct LeadingWhitespaceRule: ConfigurationProviderRule, SourceKitFreeRule {
+extension String {
+    private func countOfLeadingCharactersInSet(characterSet: NSCharacterSet) -> Int {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift]: why did you re-add this? It's already available.

File: Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift [hunk 2]
```diff
@@ -9,7 +9,23 @@
...
+        var count = 0
+        for char in utf16 {
+            if !characterSet.characterIsMember(char) {
+                break
+            }
+            count += 1
+        }
+        return count
+    }
+    private func leadingNewlineAndWhiteSpaceCount() -> Int? {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift]: if we add this (which I'm not sure we need to), we should use it where we're currently using the expanded form (e.g. `validateFile(_:)`).

File: Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift [hunk 3]
```diff
@@ -36,4 +53,22 @@ public struct LeadingWhitespaceRule: ConfigurationProviderRule, SourceKitFreeRul
...
     }
+
+    public func correctFile(file: File) -> [Correction] {
+        guard let spaceCount = file.contents.leadingNewlineAndWhiteSpaceCount()
+            where spaceCount != 0  else {
+            return []
+        }
+        let region = file.regions().filter {
+            $0.contains(Location(file: file.path, line: max(file.lines.count, 1)))
+            }.first
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift]: should be indented one level less.

File: Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift [hunk 4]
```diff
@@ -36,4 +53,22 @@ public struct LeadingWhitespaceRule: ConfigurationProviderRule, SourceKitFreeRul
...
+        if region?.isRuleDisabled(self) == true {
+            return []
+        }
+        file.write(file.contents.substringFromIndex(
+                file.contents.startIndex.advancedBy(spaceCount)))
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/LeadingWhitespaceRule.swift]: indentation is odd here.

**Comment by jpsim** [comment]
Thanks for the PR @masters3d! I just had a few comments, otherwise looks good!

Also, please add a changelog entry following the guidelines in the CONTRIBUTING.md document. Thanks!

**Comment by masters3d** [comment]
nice! looking forward to contributing more.

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,9 @@
 
 ##### Enhancements
 
+* LeadingWhitespaceRule is now auto correctable.
```
Comment (jpsim) [inline CHANGELOG.md]: missing two trailing spaces, according to the guidelines in contributing.md

**Comment by jpsim** [comment]
👍 looks good, other than the whitespace thing. Thanks!

**Pull request merged**